### PR TITLE
Assemble final font

### PR DESCRIPTION
**Pull request by rsheeter**
If you compile a font and try to ttx it sadly it blows up because we don't have maxp yet. Soon!

I would like to merge this and then grow the set of tables over the next couple of PRs as we drive towards #138.

### CODE REVIEW

File: fontbe/src/font.rs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+struct FontWork {}
+
+pub fn create_font_work() -> Box<BeWork> {
+    Box::new(FontWork {})
+}
+
+// TODO the tables have the tag, but I'm not sure how to access it
+const TABLES_TO_MERGE: &[(WorkId, Tag)] = &[
+    (WorkId::Cmap, Tag::new(b"cmap")),
```
Comment (anthrotype) [inline fontbe/src/font.rs:20]: You're missing perhaps the most important tables of all, maxp, which defines _how many glyphs_ are in the font :)

File: fontbe/src/font.rs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    (WorkId::Hmtx, Tag::new(b"hmtx")),
+    (WorkId::Glyf, Tag::new(b"glyf")),
+    (WorkId::Loca, Tag::new(b"loca")),
```
Comment (anthrotype) [inline fontbe/src/font.rs:23]: needless to say, hmtx won't work without hhea, and loca (thus glyf) won't work without head (likely other tables too depend on head too), but those are easy if you just want something that works and don't care about the particular values in it. OS/2 might be needed for windows I think (just use the same vertical metrics values as hhea).

Comment (anthrotype) [reply fontbe/src/font.rs:23]: oh, name table seems to be required by OTS (I just tried to strip it from a font and it complained)

Comment (rsheeter) [reply fontbe/src/font.rs:23]: Yup. I think I want to merge this then add more tables as we progress through #138.

Comment (rsheeter) [reply fontbe/src/font.rs:23]: To paraphrase me, this puts the pattern to create a working font in place. It doesn't yet fully do it due to missing tables. My current expectation is if we hit the set identified in #138 we're good.

File: fontbe/src/orchestration.rs [hunk 3]
```diff
@@ -210,12 +213,26 @@ where
     T::read(FontData::new(&buf)).unwrap()
 }
 
-fn raw_hmtx_from_file(file: &Path) -> RawHmtx {
+pub struct Bytes {
```
Comment (anthrotype) [inline fontbe/src/orchestration.rs:216]: you have an identical `Raw` struct in fontbe::hmtx module...

File: fontbe/src/hmtx.rs [hunk 4]
```diff
@@ -8,10 +8,10 @@ use write_fonts::{
 
 use crate::{
     error::Error,
-    orchestration::{BeWork, Context},
+    orchestration::{BeWork, Bytes, Context},
 };
 
-pub struct RawHmtx {
+pub struct Raw {
```
Comment (anthrotype) [inline fontbe/src/hmtx.rs:14]: this is no longer used, is it?

**Review by anthrotype — APPROVED** [review]

File: fontbe/src/font.rs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    /// Glue binary tables into a font
+    fn exec(&self, context: &Context) -> Result<(), Error> {
+        // Lets go right ahead and believe those bytes are a font
+        let mut builder = FontBuilder::default();
+
+        // A fancier implementation would mmap the files. We basic.
+        for (work_id, tag) in TABLES_TO_MERGE.iter().cloned() {
+            let bytes = context.read_raw(work_id).map_err(Error::IoError)?;
+            builder.add_table(tag, bytes);
+        }
```
Comment (cmyr) [inline fontbe/src/font.rs:36]: Merely for the sake of conversation, I would probably write this as,

```suggestion
        for (work_id, tag) in TABLES_TO_MERGE {
            let bytes = context.read_raw(*work_id).map_err(Error::IoError)?;
            builder.add_table(*tag, bytes);
        }
```

🤷

File: fontbe/src/orchestration.rs [hunk 6]
```diff
@@ -210,12 +213,26 @@ where
...
+    pub(crate) fn new(buf: Vec<u8>) -> Bytes {
+        Bytes { buf }
+    }
+
+    pub fn get(&self) -> &[u8] {
+        &self.buf
+    }
+}
+
+fn raw_from_file(file: &Path) -> Bytes {
```
Comment (cmyr) [inline fontbe/src/orchestration.rs:230]: does this want to be `Bytes::read_from_file` or something?

**Review by cmyr — APPROVED** [review]
Some very ignorable comments inline, lgtm

File: fontbe/src/font.rs (hunk 5 above)
Comment (rsheeter) [reply fontbe/src/font.rs:36]: I had that and thought it less readable. Do you find the opposite?

File: fontbe/src/orchestration.rs (hunk 3 above)
Comment (rsheeter) [reply fontbe/src/orchestration.rs:216]: I have 2, need 1, and would prefer ot have 0 rofl. Will fix.

File: fontbe/src/hmtx.rs (hunk 4 above)
Comment (rsheeter) [reply fontbe/src/hmtx.rs:14]: Nope! Deleted.

File: fontbe/src/orchestration.rs (hunk 3 above)
Comment (rsheeter) [reply fontbe/src/orchestration.rs:216]: Keeping this one.

File: fontbe/src/orchestration.rs (hunk 6 above)
Comment (rsheeter) [reply fontbe/src/orchestration.rs:230]: That seems to complicate my macro. Will add a comment.

**Pull request merged**